### ISSUE TITLE
Assertion `box.hasRank()' failed.

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#494]
From sixtrack.

```
% cat dv.f
          SUBROUTINE KERSET(ERCODE,LGFILE,LIMITM,LIMITR)
          IMPLICIT REAL*8 (A-H,O-Z)
          INTEGER  KNTR(28)
          SAVE
          DO  I  =   1,  KOUNTE
            KNTR(I)  =  LIMITR
          END DO
          RETURN
          ENTRY KERMTR(ERCODE,LOG,MFLAG,RFLAG)
          RETURN
          END
```
```
% bbc dv.f |& head
bbc: /local/home/sjs/mirror/fir-dev-mirror/flang/lib/Lower/ConvertExpr.cpp:1331: mlir::Value {anonymous}::ExprLowering::getLBound(const Fortran::lower::SymbolBox&, unsigned int, mlir::Type): Assertion `box.hasRank()' failed.
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc dv.f
 #0 0x0000561a464ef50d llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/local/home/sjs/mirror/install/bin/bbc+0x340f50d)
 #1 0x0000561a464ed244 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x340d244)
 #2 0x0000561a464ed3b0 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x340d3b0)
 #3 0x000014ba5f4ee8a0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x128a0)
 #4 0x000014ba5e15af47 raise /build/glibc-2ORdQG/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x000014ba5e15c8b1 abort /build/glibc-2ORdQG/glibc-2.27/stdlib/abort.c:81:0
```

**Comment by jeanPerier** [comment]
Same root cause of #495, will be fixed by my #495 fix.

**Comment by jeanPerier** [comment]
Should be fixed by #501 that was merged.

**Comment by sscalpone** [comment]
Fixed!
